### PR TITLE
Another e2e beta test fix

### PR DESCRIPTION
**Pull request by danjm**
~~Line 519 of https://github.com/MetaMask/metamask-extension/compare/e2e-tests-more-reliable?expand=1#diff-b00929238ad0453812af55a40336e889 was incorrectly matching the text on line 14 of test/e2e/beta/contract-test/index.html before the contract deploy was actually complete.~~

~~By updating the text on line 14 of test/e2e/beta/contract-test/index.html, these tests should now more consistently pass.~~

The previous description of the problem solved by this PR was incorrect. The updated description of this PR is:

This PR is an attempt to fix an inconsistently failing test in `metamask-beta-ui.spec.js`. The `'calls and confirms a contract method where ETH is sent'` test block fairly regularly, but not always, fails in firefox.

Currently the attempt to fix this is adding a missing `await` and improving the error handling within the method called on line 441: `await assertElementNotPresent(webdriver, driver, By.xpath(`//li[contains(text(), 'Data')]`))`

Further changes are made to just improve test debugging.

I will leave this PR open until we get the firefox tests constistently passing.

### CODE REVIEW

File: test/e2e/beta/metamask-beta-ui.spec.js [hunk 1]
```diff
@@ -515,15 +515,15 @@ describe('MetaMask', function () {
       await driver.switchTo().window(dapp)
       await delay(regularDelayMs)
 
-      let contractStatus = await driver.findElement(By.css('#contractStatus'))
-      await driver.wait(until.elementTextMatches(contractStatus, /Deployed/))
+      let contractStatus = await driver.findElement(By.css('#contractStatus'), 10000)
```
Comment (whymarrh) [inline test/e2e/beta/metamask-beta-ui.spec.js]: Can we use `await findElement(driver, By.css(...))` as that already has the longer timeout?

Comment (whymarrh) [reply test/e2e/beta/metamask-beta-ui.spec.js]: Similarly for all instances below

**Review by whymarrh — CHANGES_REQUESTED** [review]

**Comment by whymarrh** [comment]
> Line 519 of [...] was incorrectly matching the text on line 14 of [...] before the contract deploy was actually complete

How do we know this is the case? I don't _think_ that should happen given that the regex isn't flagged as case-insensitive (that is, it _is_ case-sensitive) and where it is `/Deployed/` and not `/deployed/` the regex shouldn't match the original text.

**Comment by danjm** [comment]
> How do we know this is the case? I don't think that should happen given that the regex isn't flagged as case-insensitive (that is, it is case-sensitive) 

@whymarrh yeah, when I wrote that I was thinking that regex is case-sensitive by default. Sometimes I forget how regex works after midnight.

I have completely updated the PR description and commits.

**Comment by whymarrh** [comment]
> I will leave this PR open until we get the firefox tests constistently passing.

I guess we still have a bit of digging to do—`test-e2e-beta-firefox` is failing with the new timeout

File: test/e2e/beta/contract-test/contract.js [hunk 2]
```diff
@@ -38,18 +38,19 @@ const transferTokens = document.getElementById('transferTokens')
...
       gas: '4700000',
     }, function (e, contract) {
-      console.log(e, contract)
-      if (typeof contract.address !== 'undefined') {
+      if (e) {
+        throw e
+      } else if (typeof contract.address !== 'undefined') {
```
Comment (whymarrh) [inline test/e2e/beta/contract-test/contract.js]: I don't think this `else` is needed since we're throwing in `if (e)`

**Review by whymarrh — APPROVED** [review]

**Pull request merged**